RIOT-OS/RIOT
# build system: Restructure dependency resolution

### Contribution description

This PR adds `FEATURES_REQUIRED_ONE_OUT_OF` to the build system if one (or more) of a set of features is needed, no matter which. This is resolved after the classic `FEATURES_REQUIRED`, so that additional features are only not any of the other dependencies have already pulled in a viable features.

As proof of concept, `tests/driver_ws281x` has been updated to use this.

### Testing procedure

- `tests/driver_ws281x` should still have the same output for `make info-boards-supported`
- Compilation of no other app is broken (Murdock will check this)

### Issues/PRs references

Alternative to https://github.com/RIOT-OS/RIOT/pull/13343
